include/votca/xtp/segment.h

```diff
@@ -0,0 +1,199 @@
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+#ifndef __VOTCA_XTP_SEGMENT_H
+#define	__VOTCA_XTP_SEGMENT_H
+
+#include <map>
+#include <vector>
+
+#include <votca/tools/vec.h>
+
+namespace votca { namespace xtp {
+
+class Atom;
+class PolarSite;
+class APolarSite;
+class Fragment;
+class SegmentType;
+class Topology;
+class Molecule;  
+
+class Segment
+{
+public:
+
+    Segment(int id, std::string name);
+    Segment(Segment *stencil);
+   ~Segment();
+
+    const int       &getId() { return _id; }
+    const std::string    &getName() { return _name; }
+
+    const tools::vec       &getPos() const { return _CoM; }
+    void             setPos(tools::vec pos) { _CoM = pos; }
+    // This gets the center of mass from the MD positions of the atoms
+    void             calcPos();
+    void             TranslateBy(const tools::vec &shift);
+    
+    void            calcApproxSize();
+    double          getApproxSize(){return _approxsize;}
+
+    void             setHasState(bool yesno, int state);
+    bool             hasState(int state);
+
+    double           getOcc(int e_h_s_t);
+    void             setOcc(double occ, int e_h_s_t);
+
+    // state: -1 electron +1 hole +2 singlet +3 triplet
+  
+    /// Following notation can be observed in: 
+    /// [1. Victor, R. et al. Microscopic Simulations of Charge Transport in Disordered Organic Semiconductors. J. Chem. Theory Comput. 7, 3335–3345 (2011).] 
+    /// Labeling of the following methods follows the following semantics:
+    /// U - Energy 
+    /// n - neutral geometry
+    /// N - neutral state
+    /// c - charged geometry
+    /// C - charged state
+    /// x - excited geometry
+    /// X - excited state
+
+    /// UcC - UnN
+    void             setU_cC_nN(double dU, int state);
+    /// UnN - UnN
+    void             setU_nC_nN(double dU, int state);
+    /// UcN - UcC
+    void             setU_cN_cC(double dU, int state);
+    /// UxX - UnN
+    void             setU_xX_nN(double dU, int state);
+    /// UnX - UnN
+    void             setU_nX_nN(double dU, int state);
+    /// UxN - UxX
+    void             setU_xN_xX(double dU, int state);
+    const double    &getU_cC_nN(int state);
+    const double    &getU_nC_nN(int state);
+    const double    &getU_cN_cC(int state);
+    const double    &getU_xX_nN(int state);
+    const double    &getU_nX_nN(int state);
+    const double    &getU_xN_xX(int state);
+    double           getSiteEnergy(int state);
+
+    double           getEMpoles(int state);
+    void             setEMpoles(int state, double energy);
```

Comment:
why does segment have this functionality and what does it do?

Reply:
No idea. We will look at it during the refactoring stage.